### PR TITLE
Added stm32l476 support, trivial kernel source code cleanup

### PR DESCRIPTION
**Pull request by Kochise**
Cleaned things into two separate commits

### CODE REVIEW

**Comment by jserv** [comment]
@Kochise, Thanks again for sending pull request. Could you please use "git rebase -i` to rework these commits so that they can look cleaner and functioned?

**Comment by Kochise** [comment]
Thanks for the guidance, since I'm not a git expert.

**Comment by Kochise** [comment]
Done it, doesn't seems to have changed anything. More advices ?

File: board/nucleol476rg/board.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2016 The F9 Microkernel Project. All rights reserved.
```
Comment (lulouie) [inline board/nucleol476rg/board.c]: Please change the copyright year to `2017`

File: include/platform/stm32l476/exti.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#include <platform/stm32l476/gpio.h>
+
+/* EXTI mode */
+#define EXTI_INTERRUPT_MODE				0x0
+#define EXTI_EVENT_MODE					0x4
+
+/* EXTI trigger type */
+#define EXTI_RISING_TRIGGER				0x8
+#define EXTI_FALLING_TRIGGER			0xc
+#define EXTI_RISING_FALLING_TRIGGER		0x10
```
Comment (lulouie) [inline include/platform/stm32l476/exti.h]: Please help to modify the style to make it consist.

```
#define foo      0x0
#define bar      0x.4
...etc
```

File: include/platform/stm32l476/gpio.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#define     af_i2c1             AF4
+#define     af_i2c2             AF4
+#define     af_i2c3             AF4
+#define     af_spi1             AF5
+#define     af_spi2             AF5
+#define     af_spi5             AF5
+#define     af_spi3             AF6
+#define     af_usart1           AF7
+#define     af_usart2           AF7
+#define     af_usart3		AF7
```
Comment (lulouie) [inline include/platform/stm32l476/gpio.h]: same here

File: include/platform/stm32l476/syscfg.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+{
+	*SYSCFG_MEMRMP = 0x0;
+}
+
+static inline void memory_remap_sysflash(void)
+{
+	*SYSCFG_MEMRMP = 0x1;
+}
+
+
```
Comment (lulouie) [inline include/platform/stm32l476/syscfg.h]: duplicate white line

File: include/platform/stm32l476/usart.h [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#define USART_TXE	((uint16_t) (0x1 << 7))
+#define USART_TC	((uint16_t) (0x1 << 6))
+#define USART_RXNE	((uint16_t) (0x1 << 5))
+#define USART_IDLE	((uint16_t) (0x1 << 4))
+#define USART_ORE	((uint16_t) (0x1 << 3))
+#define USART_NF	((uint16_t) (0x1 << 2))
+#define USART_FE	((uint16_t) (0x1 << 1))
+#define USART_PE	((uint16_t) (0x1 << 0))
+
+
```
Comment (lulouie) [inline include/platform/stm32l476/usart.h]: duplicate line

File: kernel/interrupt.c [hunk 6]
```diff
@@ -86,8 +88,10 @@ static struct user_irq *user_irq_queue_pop(void)
 
 static void user_irq_queue_delete(int irq)
 {
-	struct user_irq *uirq = user_irqs[irq];
-	for (struct user_irq **iter = &user_irq_queue.head ; *iter != NULL ;
+	struct user_irq **iter, *uirq;
+
+	uirq = user_irqs[irq];
+	for (iter = &user_irq_queue.head ; *iter != NULL ;
```
Comment (lulouie) [inline kernel/interrupt.c]: Please use C99 style as before

File: kernel/interrupt.c [hunk 7]
```diff
@@ -98,14 +102,18 @@ static void user_irq_queue_delete(int irq)
 
 static inline void user_irq_reset_all(void)
 {
-	for (int i = 0 ; i < IRQn_NUM ; i++)
+	int i;
+
+	for (i = 0 ; i < IRQn_NUM ; i++)
```
Comment (lulouie) [inline kernel/interrupt.c]: ditto

File: kernel/kprobes.c [hunk 8]
```diff
@@ -54,13 +54,16 @@ void kplist_del(struct kprobe *kp)
...
+	int ret;
+
 	kp->addr = (void *) ((uint32_t) kp->addr & ~(1UL));
 	if (is_thumb32(*(uint16_t *) kp->addr))
 		kp->step_addr = kp->addr + 4;
 	else
 		kp->step_addr = kp->addr + 2;
 
-	if (kprobe_arch_add(kp) < 0)
+	ret = kprobe_arch_add(kp);
```
Comment (lulouie) [inline kernel/kprobes.c]: no need to seperate this, since it will return int itself.

**Comment by lulouie** [comment]
Also, there is no need for rebase / sqashing commit, since when finish the PR, member can squash via GitHub UI, thanks!

**Comment by Kochise** [comment]
"for (int i = 0;" is considered bad coding practices, the declaration and affectation should be separated. All members and local variables should be placed at one location only to ease maintenance and avoid scope conflicts. Btw I separated these into two commits, the one regarding C99 styling is isolated, you just have to ditch it out if the changes really doesn't fits your taste.

I separated the 'ret =' because it is considered best practice and easier to debug to have a  variable holding the value. Clever 'optimizations' like this should be left to the compiler.

Copyright year is 2016 because the code changes were made last year, but only submitted this year. If that is really important, I will do so, just confirm.

About rebasing/sqashing, I have no much idea what it serves as a purpose and how to do it.

Thanks for your reviews and feedback.

Your truly.

**Comment by lulouie** [comment]
@Kochise friend, I think we both want this thing better, I must apologize for that I didn't catch up the previous PR discussion. 

Your point about these variable make sense, and I think the copyright year probelm, if that is a new code (e.g. new board specific) need to move to 2017, others don't need to change.

Again, thanks for your helping and contribute.
Louie.

**Comment by Kochise** [comment]
No problem, I'm rather happy to contribute to this nice project of yours :)

Will change copyright date to 2017 of the stm32l4 files.
Will remove kernel code modifications to leave it as is.

Sorry for my rants about coding style. This is a very sensitive subject to me.

**Pull request closed without merging**